consul/src/main/scala/io/buoyant/consul/v1.scala

```diff
@@ -121,6 +126,128 @@ package object v1 {
     }
   }
 
+  object KvApi {
+    def apply(c: Client): KvApi = new KvApi(c, s"/$versionString")
+  }
+
+  class KvApi(
+    client: Client,
+    uriPrefix: String,
+    backoffs: Stream[Duration] = Backoff.exponentialJittered(1.milliseconds, 5.seconds),
+    stats: StatsReceiver = DefaultStatsReceiver
+  ) extends Closable {
+    def close(deadline: Time) = client.close(deadline)
+
+    val kvPrefix = s"$uriPrefix/kv"
+
+    private[this] val infiniteRetryFilter = new RetryFilter[http.Request, http.Response](
```

Comment:
infiniteRetryFilter and retryClient could be factored into the `Api` common trait